vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts

```diff
@@ -0,0 +1,304 @@
+            )
+            .filter
+            (
+                function(value : string, i : number) { return i % 3 === 0; } // Every 0th, 4th, etc item will be a value name AKA the SDK version. The rest will be REGTYPE and REGHEXVALUE.
+            );
+        }
+    }
+
+    /**
+     *
+     * @returns Returns '' if no conflicting version was found on the machine.
+     * Returns the existing version if a global install with the requested version already exists.
+     * OR: If a global install exists for the same band with a higher version.
+     * For non-windows cases: In Mac the installer is always shown so that will show users this. For Linux, it's handled by the distro specific code.
+     */
+    public async GlobalWindowsInstallWithConflictingVersionAlreadyExists(requestedVersion : string) : Promise<string>
+    {
+        // Note that we could be more intelligent here and consider only if the SDKs conflict within an architecture, but for now we won't do this.
+        const sdks : Array<string> = await this.getGlobalSdkVersionsInstalledOnMachine();
+        for (const sdk of sdks)
+        {
+            if
+            ( // Side by side installs of the same major.minor and band can cause issues in some cases. So we decided to just not allow it unless upgrading to a newer patch version.
+              // The installer can catch this but we can avoid unnecessary work this way,
+              // and for windows the installer may never appear to the user. With this approach, we don't need to handle installer error codes.
+                Number(this.versionResolver.getMajorMinor(requestedVersion)) === Number(this.versionResolver.getMajorMinor(sdk)) &&
+                Number(this.versionResolver.getFeatureBandFromVersion(requestedVersion)) === Number(this.versionResolver.getFeatureBandFromVersion(sdk)) &&
+                Number(this.versionResolver.getFeatureBandPatchVersion(requestedVersion)) <= Number(this.versionResolver.getFeatureBandPatchVersion(sdk))
+            )
+            {
+                return sdk;
+            }
+        }
+
+        return '';
+    }
+
+    /**
+     *
+     * @returns an array containing fully specified / specific versions of all globally installed sdks on the machine in windows for 32 and 64 bit sdks.
+     */
+    public async getGlobalSdkVersionsInstalledOnMachine() : Promise<Array<string>>
+    {
+        let sdks: string[] = [];
+
+
+        if (os.platform() === 'win32')
+        {
+            const sdkInstallRecords64Bit = 'HKEY_LOCAL_MACHINE\\SOFTWARE\\dotnet\\Setup\\InstalledVersions\\x64\\sdk';
+            const sdkInstallRecords32Bit = sdkInstallRecords64Bit.replace('x64', 'x86');
+
+            const queries = [sdkInstallRecords32Bit, sdkInstallRecords64Bit];
+            for ( const query of queries)
+            {
+                try
+                {
+                    const registryQueryCommand = path.join(`${process.env.SystemRoot}`, `System32\\reg.exe`);
```

Comment:
reg.exe could be disabled, so we should handle that situation

Reply:
I fixed it so it works, though it failed with a warning to add .NET to the path. The installer does this for us, so I stopped running that code for global installs for win and mac.